Configs/.local/share/bin/cliphist.sh

```diff
@@ -1,86 +1,87 @@
-y_mon=$((y_mon * 100 / monitor_scale))
-#? monitor position
-x_pos=$(hyprctl -j monitors | jq '.[] | select(.focused==true) | .x')
-y_pos=$(hyprctl -j monitors | jq '.[] | select(.focused==true) | .y')
-#? cursor position
-x_cur=$(hyprctl -j cursorpos | jq '.x')
-y_cur=$(hyprctl -j cursorpos | jq '.y')
-# Ignore position
- x_cur=$(( x_cur - x_pos))
- y_cur=$(( y_cur - y_pos))
-#Limiting
-# Multiply before dividing to avoid losing precision due to integer division
-clip_w=$(( x_mon*clip_w/100 ))
-clip_h=$(( y_mon*clip_h/100 ))
-max_x=$((x_mon - clip_w - 5 )) #offset of 5 for gaps
-max_y=$((y_mon - clip_h - 15 )) #offset of 15 for gaps
-x_cur=$((x_cur - x_offset))
-y_cur=$((y_cur - y_offset))
-# 
-x_cur=$(( x_cur < min_x ? min_x : ( x_cur > max_x ? max_x :  x_cur)))
-y_cur=$(( y_cur < min_y ? min_y : ( y_cur > max_y ? max_y :  y_cur)))
+x_offset=${clipXoffset:--15} #* Cursor spawn position on clipboard
+y_offset=${clipYoffset:-210} #* To point the Cursor to the 1st and 2nd latest word
 
-pos="window {location: north west; x-offset: ${x_cur}px; y-offset: ${y_cur}px;}" #! I just Used the old pos function
-#pos="window {location: $y_rofi $x_rofi; $x_offset $y_offset}" 
+# Monitor resolution , scale and rotation
+eval "$(hyprctl monitors -j | jq -r \
+    ' .[] | select(.focused==true) | 
+ (if (.transform | (. % 2) == 1) then
+   {monWidth: (.height / .scale | floor), monHeight: (.width / .scale | floor)}
+ else
+   {monWidth: (.width / .scale | floor), monHeight: (.height / .scale | floor)}
+ end) as $dims |
+"export monName=\(.name);
+export monWidth=\($dims.monWidth);
+export monHeight=\($dims.monHeight);
+export monXpos=\(.x | floor);
+export monYpos=\(.y | floor);
+"')"
 
-# read hypr theme border
+# Level 1 layers  e.g namespace for  waybar, for now just waybar \\ ensures that we can get a good boundary value
+# TODO Handle if we got multple waybar location
+wBar=0
+hBar=0
+eval "$(hyprctl layers -j | jq -r --arg mon "$monName" '.[$mon].levels | .[] | .[] | select(.namespace == "waybar") | if .h < .w then "export export hBar=$((\(.h) + $hBar ))" else "export wBar=$((\(.w) + $wBar ))" end')"
 
-wind_border=$(( hypr_border * 3/2 ))
-elem_border=`[ $hypr_border -eq 0 ] && echo "5" || echo $hypr_border`
-r_override="window {border: ${hypr_width}px; border-radius: ${wind_border}px;} entry {border-radius: ${elem_border}px;} element {border-radius: ${elem_border}px;}"
+#  Cursor position filtered by Monitor stats
+eval "$(hyprctl cursorpos -j | jq -r \
+    --argjson x_offset "$x_offset" \
+    --argjson y_offset "$y_offset" \
+    --argjson monXpos "$monXpos" \
+    --argjson monYpos "$monYpos" \
+    '"
+export curXpos=\(.x - $monXpos - $x_offset )
+export curYpos=\(.y - $monYpos - $y_offset)
+"')"
 
+# TODO Dynamically follow the rofi scaling if .rasi changes
+# Limit board location to monitor
+wBoard=${wBoard:-300}
+hBoard=${hBoard:-550}
+xBound=$((monWidth - wBoard - wBar))
+yBound=$((monHeight - hBoard - hBar))
```

Comment:
This part is our magic number. For now, this is not dynamic.

Reply:
> This part is our magic number. For now, this is not dynamic.

whats the `wBoard` and `hBoard` value for??

Reply:
The width and height of the pasteboard

![image](https://github.com/prasanthrangan/hyprdots/assets/53417443/77026a26-8943-4a90-b27d-eb77cc7ea5c4)

It's blank for now, but ready to be overwritten

Reply:
Added a logic to just save the scaling
![image](https://github.com/prasanthrangan/hyprdots/assets/53417443/4ca14f51-80fc-47f6-9a5f-2e525c278ee5)